SwiftGit2/Repository.swift

```diff
@@ -362,6 +362,27 @@ final public class Repository {
 		git_remote_free(pointer)
 		return Result.success(value)
 	}
+	
+	public func fetch(_ remote: Remote) -> Result<(), NSError> {
+		var pointer: OpaquePointer? = nil
+		let result0 = git_remote_lookup(&pointer, self.pointer, remote.name)
+		guard result0 == GIT_OK.rawValue else {
```

Comment:
I don't really like `result0` and `result`, but I don't know if we should assume the lookup succeeds (maybe the remote has been deleted in the meantime?)

Reply:
I think you're right to not assume that the lookup succeeds. I think I've elsewhere named the variable something like `lookupResult`. It would probably be ideal to return a more specific error (c.f. #74) and use an enum case for when the remote no longer exists, but I don't think that's required for now.

We could probably add a helper that's used here and in `remote(named:)` above that actually performed the lookup and gives you access to the pointer for the remote inside a block. If it also freed the remote, that would probably also be helpful—it looks like you're missing a `git_remote_free` here.

Or add a type that wraps `OpaquePointer` with a freeing function to call in `deinit`. Then the helper could return a `Result<SmartPointer, NSError>` and this method could use `flatMap`. 🤷‍♂️

Reply:
Yes. So the think I like a lot about a class-based approach with deinit is that you don't need to think so much about memory management: you always pass in a pointer and free it in deinit. I think having some kind of system for this would be nice.

Reply:
(also: good catch). I just pushed some new changes and a helper.